torchao/quantization/quant_api.py

```diff
@@ -684,14 +708,10 @@ def gemlite_uintx_weight_only(
     return _get_linear_subclass_inserter(apply_fn)
 
 
-def int4_weight_only(
-    group_size=128,
-    layout=TensorCoreTiledLayout(inner_k_tiles=8),
-    use_hqq=False,
-    zero_point_domain=None,
-):
+@dataclass
+class int4_weight_only(ao_base_workflow_config):
```

Comment:
What do you think about having `int4_weight_only` be a function that returns a config instead:
```
def int4_weight_only(group_size: int):
    return AffineQuantizationConfig(
        weight_dtype=torch.int4,
        weight_group_size=group_size,
        ...
        activation_dtype=None,
    )
```

Then we only need one config class for AQT in general instead of one per quant setting. Also the user can still call `quantize_(model, int4_weight_only(group_size=32))` exactly the same way as before.

Reply:
Update: decided to do 1 class per quant setting offline